service/featuregate/gates.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregate
+
+import (
+	"fmt"
+)
+
+// Gates is the interface used to query the status of feature gates. It can be
+// implemented by a read-only copy of the feature gate registry.
+type Gates interface {
+	// IsEnabled returns true if a registered feature gate is enabled and false otherwise.
+	IsEnabled(id string) bool
+
+	// List returns a slice of copies of all registered Gates.
+	List() []Gate
```

Comment:
Is this needed on the intterace? Looks like it is call in tests only and only for the concrete `registry` struct. Maybe move there?